optimizely/optimizely_user_context.py

```diff
@@ -114,3 +148,143 @@ def as_json(self):
             'user_id': self.user_id,
             'attributes': self.get_user_attributes(),
         }
+
+    def set_forced_decision(self, OptimizelyDecisionContext, OptimizelyForcedDecision):
+        """
+        Sets the forced decision for a given decision context.
+
+        Args:
+            OptimizelyDecisionContext: a decision context.
+            OptimizelyForcedDecision: a forced decision.
+
+        Returns:
+            True if the forced decision has been set successfully.
+        """
+        config = self.client.get_optimizely_config()
+
+        if self.client is None or config is None:
+            self.log.logger.error(OptimizelyDecisionMessage.SDK_NOT_READY)
+            return False
+
+        context = OptimizelyDecisionContext
+        decision = OptimizelyForcedDecision
+
+        with self.lock:
+            self.forced_decisions[context] = decision
+
+        return True
+
+    def get_forced_decision(self, OptimizelyDecisionContext):
+        """
+        Gets the forced decision (variation key) for a given decision context.
+
+        Args:
+            OptimizelyDecisionContext: a decision context.
+
+        Returns:
+            A variation key or None if forced decisions are not set for the parameters.
+        """
+        config = self.client.get_optimizely_config()
+
+        if self.client is None or config is None:
+            self.log.logger.error(OptimizelyDecisionMessage.SDK_NOT_READY)
+            return None
+
+        forced_decision_key = self.find_forced_decision(OptimizelyDecisionContext)
+
+        return forced_decision_key if forced_decision_key else None
+
+    def remove_forced_decision(self, OptimizelyDecisionContext):
+        """
+        Removes the forced decision for a given flag and an optional rule.
+
+        Args:
+            OptimizelyDecisionContext: a decision context.
+
+        Returns:
+            Returns: true if the forced decision has been removed successfully.
+        """
+        config = self.client.get_optimizely_config()
+
+        if self.client is None or config is None:
+            self.log.logger.error(OptimizelyDecisionMessage.SDK_NOT_READY)
+            return False
+
+        with self.lock:
+            if self.forced_decisions[OptimizelyDecisionContext]:
```

Comment:
It will cause KeyError if `OptimizelyDecisionContext` is not found in `forced_decisions`
```suggestion
            if OptimizelyDecisionContext in self.forced_decisions.keys()
```